agent/watch_handler.go

```diff
@@ -87,3 +90,72 @@ func makeWatchHandler(logOutput io.Writer, handler interface{}) watch.HandlerFun
 	}
 	return fn
 }
+
+func makeHTTPWatchHandler(logOutput io.Writer, config *watch.HttpHandlerConfig) watch.HandlerFunc {
+	logger := log.New(logOutput, "", log.LstdFlags)
+
+	fn := func(idx uint64, data interface{}) {
+		trans := cleanhttp.DefaultTransport()
+
+		// Skip SSL certificate verification if TLSSkipVerify is true
+		if trans.TLSClientConfig == nil {
+			trans.TLSClientConfig = &tls.Config{
+				InsecureSkipVerify: config.TLSSkipVerify,
+			}
+		} else {
+			trans.TLSClientConfig.InsecureSkipVerify = config.TLSSkipVerify
+		}
+
+		// Create the HTTP client.
+		httpClient := &http.Client{
+			Timeout:   config.Timeout,
+			Transport: trans,
+		}
```

Comment:
Use context for proper cancellation of requests that have timed out.  Create a context [WithTimeout](https://golang.org/pkg/context/#WithTimeout), and wire that into the http request below via https://golang.org/pkg/net/http/#Request.WithContext